lib/cartopy/mpl/feature_artist.py

```diff
@@ -202,20 +217,67 @@ def draw(self, renderer, *args, **kwargs):
 
             stylised_paths.setdefault(style, []).extend(geom_paths)
 
-        transform = ax.projection._as_mpl_transform(ax)
+        return stylised_paths
 
-        # Draw one PathCollection per style. We could instead pass an array
-        # of style items through to a single PathCollection, but that
-        # complexity does not yet justify the effort.
-        for style, paths in stylised_paths.items():
-            style = style_finalize(dict(style))
-            # Build path collection and draw it.
-            c = matplotlib.collections.PathCollection(paths,
-                                                      transform=transform,
-                                                      **style)
-            c.set_clip_path(ax.patch)
-            c.set_figure(ax.figure)
-            c.draw(renderer)
+    def get_geometry(self):
+        ax = self.axes
+        extent = None
+        if ax is not None:
+            transform = ax.projection._as_mpl_transform(ax)
+            feature_crs = self._feature.crs
+            # Get geometries that we need to draw.
+            try:
+                extent = ax.get_extent(feature_crs)
+            except ValueError:
+                warnings.warn('''Unable to determine extent.
+                              Defaulting to global.''')
```

Comment:
This warning should definitely not have a newline and a bunch of whitespace in it. Use plain strings, not a multiline string.